app/Http/Controllers/UpdateController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+
+namespace App\Http\Controllers;
+
+use App\Configs;
+use App\Metadata\GitHubFunctions;
+use App\Response;
+use Exception;
+
+/**
+ * Class UpdateController
+ * @package App\Http\Controllers
+ */
+class UpdateController extends Controller
+{
+
+	private $gitHubFunctions;
+
+	/**
+	 * @param GitHubFunctions $gitHubFunctions
+	 */
+	public function __construct(GitHubFunctions $gitHubFunctions)
+	{
+		$this->gitHubFunctions = $gitHubFunctions;
+	}
+
+	/**
+	 * This requires a php to have a shell access.
+	 * This method execute the update (git pull).
+	 *
+	 * @return array|string
+	 */
+	public function do()
+	{
+
+		if (Configs::get_value('allow_online_git_pull', '0') == '0') {
+			return Response::error('Online updates are not allowed.');
+		}
+		if (!function_exists('exec')) {
+			return Response::error('exec is not available');
+		}
+		if (!is_executable('../.git')) {
+			return Response::error('../.git is not executable, check the permissions');
+		}
+
+
+		try{
+			$up_to_date = $this->gitHubFunctions->is_up_to_date();
+		}
+		catch (Exception $e)
+		{
+			return Response::error($e->getMessage());
+		}
+
+		if ($up_to_date)
+		{
+			$output = [];
+			chdir('../');
```

Comment:
Don't you need to return to the original working directory when done?
If so, wouldn't it be easier for this chdir be part of the `command` (`cd .. && git/php`)?

Reply:
We don't need to return to the original working directory in this specific case.
I prefer to use the `chdir` of php to allow a failure before actually trying the command. :)
The less I do in `exec` the happier I am.